Objective-J/Runtime.js

```diff
@@ -1020,11 +1020,26 @@ GLOBAL(method_getName) = function(/*Method*/ aMethod)
     return aMethod.method_name;
 }
 
+// FIXME: This function is deprecated and should be removed in a future release
+// Please use 'method_copyReturnType' or 'method_copyArgumentType'
 GLOBAL(method_getTypes) = function(/*Method*/ aMethod)
 {
+    console.warn("Runtime function 'method_getTypes' is deprecated. Please use function 'method_copyReturnType' or 'method_copyArgumentType'");
     return aMethod.method_types;
 }
 
+GLOBAL(method_copyReturnType) = function(/*Method*/ aMethod)
+{
+    var types = aMethod.method_types;
+    return types ? types[0] : NULL;
+}
+
+GLOBAL(method_copyArgumentType) = function(/*Method*/ aMethod, /*unsigned int*/ index)
```

Comment:
Is it compliant with cocoa ? I believe in cocoa the first and second argument are types of the receiver and the selector